pkg/clients/aws.go

```diff
@@ -264,7 +266,31 @@ func UseProviderSecret(ctx context.Context, data []byte, profile, region string)
 
 // UsePodServiceAccount assumes an IAM role configured via a ServiceAccount.
 // https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
-func UsePodServiceAccount(ctx context.Context, _ []byte, _, region string) (*aws.Config, error) {
+func UsePodServiceAccount(ctx context.Context, _ []byte, rolearn string, _, region string) (*aws.Config, error) {
+	// assumeRole
+	if rolearn != "" {
```

Comment:
Having `if rolearn == ""` and returning early feels more in-line with our [conventions](https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#return-early).